src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -594,6 +632,48 @@ public static void checkCompleted(CompletionStage<?> stage) {
                                                ff.join()));
   }
 
+  /**
+   * Composes multiple stages into another stage using a function.
+   *
+   * @param function the combining function.
+   * @param stages   the stages to combine
+   * @param <T>      the type of the combining function's return value.
+   * @return a stage that is composed from the input stages using the function.
+   * @since 0.4.0
+   */
+  public static <T> CompletionStage<T> combineFutures(
+          Function<Combined, CompletionStage<T>> function, CompletionStage<?>... stages) {
+    @SuppressWarnings("unchecked") // generic array creation
+    final CompletableFuture<?>[] all = new CompletableFuture[stages.length];
+    for (int i = 0; i < stages.length; i++) {
+      all[i] = stages[i].toCompletableFuture();
+    }
+    return CompletableFuture.allOf(all)
+            .thenCompose(ignored -> function.apply(new Combined(stages)));
```

Comment:
Since this line is the only difference, maybe we could break out common utility methods for the code above this line in all of the method overloads.

Reply:
Hm, yes. Perhaps should simply be implemented something like this:

```java
 public static <T> CompletionStage<T> combineFutures(
          Function<Combined, CompletionStage<T>> function, CompletionStage<?>... stages) {
  return dereference(combine(function, stages));
}
```

Reply:
But then perhaps the entire method is not really worth including in the library at all..